pywbem/cim_operations.py

```diff
@@ -6359,6 +6359,53 @@ def ModifyInstance(self, ModifiedInstance, IncludeQualifiers=None,
         (see :term:`DSP0200`). See :ref:`WBEM operations` for a list of all
         methods performing such operations.
 
+        The properties provided in the `ModifiedInstance` parameter specify
+        the new property values for the instance to be modified.
+
+        The `PropertyList` parameter determines the set of properties that are
+        designated to be modified. The WBEM server shall reject modification
+        requests for key properties.
+
+        Pywbem sends the property values provided in the `ModifiedInstance`
+        parameter to the WBEM server as provided; it does not add any default
+        values for properties not provided but designated to be modified, nor
+        does it reduce the properties by those not designated to be modified.
+
+        DSP0200 defines the following rules for how the WBEM server modifies
+        the property values of the target instance:
+
+        * The designated new value for properties (that are designated to be
+          modified) shall be the value specified in the `ModifiedInstance`
+          parameter if it specifies a value (including `None`), else the
+          class-defined default value for the property if the class defines a
+          default value (including `None`). Otherwise, the property does not
+          have a designated new value.
+
+        * The properties are modified as follows:
+
+          * For properties designated to be modified that have a designated new
+            value, the WBEM server shall either update the property to that
+            value, or reject the request.
+          * For properties designated to be modified that do not have a
```

Comment:
This one is so general that it means nothing but that is pretty much what the spec says. However, we make the statement on line 6389 and then just below we also state that "Note that futher determinations about modifications of property values may be stated at the CIM modeling ... which is really part of the same thought.

Reply:
The sad truth is that the actual way to understand when a property can be expected to be changed is paved with all kinds of flaky rules. Pywbem being a client interface, my intention is to describe what a client needs to know to get to an understanding. I did not find an approach for that that does not allude to all these things. And no, I don't want to assume they have read and understood the various specs involved. Let's see whether we can agree on the goal and then discuss what to have in the text.

Reply:
We agreed to reduce this description to just what matters to the client, not repeating all the rules that need to be considered for someone implementing the server side. The text should include a recommendation to specify a propertylist with just the properties that are intended to get a new value.

Reply:
The following sentence in the ModifyInstance description in DSP0200 is difficult:

"If PropertyList is NULL, the properties of ModifiedInstance with values different from the current values in the instance are designated to be modified."

The difficulty is that the part about values different from current values is difficult to understand, not possible to implement reasonably in a CIMOM, and most likely not implemented in any actually existing provider. Removing this part of the sentence does not seem to change anything in the rules, because making all properties provided in the ModifiedInstance parameter those that are designated to be changed still protects the target instance from getting property values set back to their class defined default values. Which is probably what the motivation was for that sentence.

Proposal for changed sentence:

"If PropertyList is NULL, the properties of ModifiedInstance are designated to be modified."

Karl agreed to take this to the DMTF as a Mantis issue.